streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -855,6 +863,26 @@ private void checkIfUnexpectedUserSpecifiedConsumerConfig(final Map<String, Obje
         return props;
     }
 
+    /**
+     * Get a map of custom configs by removing from the originals all the Streams, Consumer, and Producer configs.
+     * Prefixed properties are also removed because they are already added by {@link #getClientPropsWithPrefix(String, Set)}.
+     * This allows to set a custom property for consumer or producer alone if specified using a prefix, or for both
+     * when no prefix is used.
+     *
+     * @param consumerPrefix the prefix used to denote consumer-only properties
+     * @param producerPrefix the prefix used to denote producer-only properties
+     * @return a map with the custom properties
+     */
+    private Map<String, Object> getClientCustomProps(final String consumerPrefix, final String producerPrefix) {
+        final Map<String, Object> props = originals();
+        props.keySet().removeAll(CONFIG.names());
+        props.keySet().removeAll(ConsumerConfig.configNames());
+        props.keySet().removeAll(ProducerConfig.configNames());
+        props.keySet().removeAll(originalsWithPrefix(consumerPrefix, false).keySet());
+        props.keySet().removeAll(originalsWithPrefix(producerPrefix, false).keySet());
```

Comment:
as above: we need to remove adminPrefix configs

Reply:
Yes, I'll do that.